keps/sig-api-machinery/20190228-priority-and-fairness.md

```diff
@@ -957,55 +961,78 @@ complete their service.
 
 ### Default Behavior
 
-There must be reasonable behavior "out of the box", and it should be
-at least a little difficult for an administrator to lock himself out
-of this subsystem.  To accomplish these things there are two levels of
-defaulting: one concerning behavior, and one concerning explicit API
-objects.
-
-The effective configuration is the union of (a) the actual API objects
-that exist and (b) implicitly generated backstop objects.  The latter
-are not actual API objects, and might not ever exist as identifiable
-objects in the implementation, but are figments of our imagination
-used to describe the behavior of this subsystem.  These backstop
-objects are implicitly present and affecting behavior when needed.
-There are two implicitly generated RequestPriority backstop objects.
-One is equivalent to the `system-top` object exhibited above, and it
-exists while there is no actual RequestPriority object with `exempt ==
-true`.  The other is equivalent to the `workload-low` object exhibited
-above, and exists while there is no RequestPriority object with
-non-exempt priority.  There are also two implicitly generated
-FlowSchema backup objects.  Whenever a request whose groups include
-`system:masters` is not matched by any actual FlowSchema object, a
-backstop equivalent to the `system-top` object exhibited above is
-considered to exist.  Whenever a request whose groups do not include
-`system:masters` is not matched by any actual FlowSchema object, the
-following backstop object is considered to exist.
+There are three default behavior questions to answer: (1) what is the
+default configuration of a cluster, (2) how does an apiserver treat
+requests before it has had a chance to read the configuration and
+ensure it is reasonable, and (3) how is a request handled if it does
+not match any FlowSchema?
+
+#### Default Cluster Configuration
+
+There is a _predefined_ set of configuration objects, and it consists
+of the objects exhibited in the
+[Example Configuration](#example-configuration) section above.
+
+Whenever a kube-apiserver starts up, it checks whether there is a
+PriorityLevelConfiguration object with `Spec.Exempt==true`.  If not
+then the apiserver creates the predefined configuration objects, the
+exempt PriorityLevelConfiguration last.  That is: for each predefined
+object, the apiserver will create that object if there is not already
+an object with the same GroupKind and name.  Note that mere existence
```

Comment:
That's because the apiserver could crash after creating some but not all.  By checking for the object that it creates last, or an expected replacement, the check does not get confused by the debris left after a crash during cluster initialization.